libcudacxx/include/cuda/std/__atomic/order.h

```diff
@@ -0,0 +1,147 @@
+inline constexpr auto memory_order_consume = memory_order::consume;
+inline constexpr auto memory_order_acquire = memory_order::acquire;
+inline constexpr auto memory_order_release = memory_order::release;
+inline constexpr auto memory_order_acq_rel = memory_order::acq_rel;
+inline constexpr auto memory_order_seq_cst = memory_order::seq_cst;
+
+#else
+
+typedef enum memory_order
+{
+  memory_order_relaxed = __mo_relaxed,
+  memory_order_consume = __mo_consume,
+  memory_order_acquire = __mo_acquire,
+  memory_order_release = __mo_release,
+  memory_order_acq_rel = __mo_acq_rel,
+  memory_order_seq_cst = __mo_seq_cst,
+} memory_order;
+
+#endif // _CCCL_STD_VER > 2017
+
+_CCCL_HOST_DEVICE inline int __stronger_order_cuda(int __a, int __b)
+{
+  int const __max = __a > __b ? __a : __b;
+  if (__max != __ATOMIC_RELEASE)
+  {
+    return __max;
+  }
+  static int const __xform[] = {__ATOMIC_RELEASE, __ATOMIC_ACQ_REL, __ATOMIC_ACQ_REL, __ATOMIC_RELEASE};
+  return __xform[__a < __b ? __a : __b];
+}
+
+_CCCL_HOST_DEVICE inline constexpr int __atomic_order_to_int(memory_order __order)
+{
+  // Avoid switch statement to make this a constexpr.
+  return __order == memory_order_relaxed
+         ? __ATOMIC_RELAXED
+         : (__order == memory_order_acquire
+              ? __ATOMIC_ACQUIRE
+              : (__order == memory_order_release
+                   ? __ATOMIC_RELEASE
+                   : (__order == memory_order_seq_cst
+                        ? __ATOMIC_SEQ_CST
+                        : (__order == memory_order_acq_rel ? __ATOMIC_ACQ_REL : __ATOMIC_CONSUME))));
+}
+
+_CCCL_HOST_DEVICE inline constexpr int __atomic_failure_order_to_int(memory_order __order)
+{
+  // Avoid switch statement to make this a constexpr.
+  return __order == memory_order_relaxed
+         ? __ATOMIC_RELAXED
+         : (__order == memory_order_acquire
+              ? __ATOMIC_ACQUIRE
+              : (__order == memory_order_release
+                   ? __ATOMIC_RELAXED
+                   : (__order == memory_order_seq_cst
+                        ? __ATOMIC_SEQ_CST
+                        : (__order == memory_order_acq_rel ? __ATOMIC_ACQUIRE : __ATOMIC_CONSUME))));
+}
+
+static_assert((is_same<underlying_type<memory_order>::type, __memory_order_underlying_t>::value),
+              "unexpected underlying type for std::memory_order");
+
+_LIBCUDACXX_END_NAMESPACE_STD
+
+_LIBCUDACXX_BEGIN_NAMESPACE_CUDA
+
+using memory_order = _CUDA_VSTD::memory_order;
+
+constexpr memory_order memory_order_relaxed = _CUDA_VSTD::memory_order_relaxed;
+constexpr memory_order memory_order_consume = _CUDA_VSTD::memory_order_consume;
+constexpr memory_order memory_order_acquire = _CUDA_VSTD::memory_order_acquire;
+constexpr memory_order memory_order_release = _CUDA_VSTD::memory_order_release;
+constexpr memory_order memory_order_acq_rel = _CUDA_VSTD::memory_order_acq_rel;
+constexpr memory_order memory_order_seq_cst = _CUDA_VSTD::memory_order_seq_cst;
```

Comment:
Should those be inline variables?
```suggestion
_LIBCUDACXX_INLINE_VAR constexpr memory_order memory_order_relaxed = _CUDA_VSTD::memory_order_relaxed;
_LIBCUDACXX_INLINE_VAR constexpr memory_order memory_order_consume = _CUDA_VSTD::memory_order_consume;
_LIBCUDACXX_INLINE_VAR constexpr memory_order memory_order_acquire = _CUDA_VSTD::memory_order_acquire;
_LIBCUDACXX_INLINE_VAR constexpr memory_order memory_order_release = _CUDA_VSTD::memory_order_release;
_LIBCUDACXX_INLINE_VAR constexpr memory_order memory_order_acq_rel = _CUDA_VSTD::memory_order_acq_rel;
_LIBCUDACXX_INLINE_VAR constexpr memory_order memory_order_seq_cst = _CUDA_VSTD::memory_order_seq_cst;
```